File: Challenge#04/challenge_04.py
def find_checksum():
    with open("Challenge#04/files_quarantine.txt", "r") as f:
        codes = f.readlines()

    count = 0  # Obtener el checksum número 33
    for code in codes:
        correct = True  # Se asume que esta correcto al principio

        password, checksum = code.strip().split("-")
        for letter in password:
            if letter in checksum:
                if password.count(letter) != 1:
                    correct = False  # Si la letra se encuentra en el checksum y existe más de una vez se cambia a False

        if correct == True:
            count += 1

        if count == 33:
            return checksum


checksum = find_checksum()
if checksum:
    print(checksum)

File: Challenge#04/challenge_04.js
const fs = require('fs')
const input = fs.readFileSync('Challenge#04/files_quarantine.txt', 'utf8').split('\n')

let count = 0 // Obtener el checksum cuando sea 33

for (let elements of input) {
    let correct = true 

    const [password, checksum] = elements.split("-")

    // Si la letra esta dentro del checksum y está más de una vez en la contraseña se asigna como false
    for (const letter of password) {
        if (checksum.includes(letter)) {
            if (password.split(letter).length - 1 !== 1) {
                correct = false
            }
        }
    }

    if (correct) {
        count++;
    }
    if (count === 33) {
        console.log(checksum);
    }
}
